File: src/main/java/be/superjoran/mint/services/BankAccountServiceImpl.java
package be.superjoran.mint.services;

import be.superjoran.common.model.DomainObjectCrudServiceSupport;
import be.superjoran.mint.dao.BankAccountDao;
import be.superjoran.mint.domain.BankAccount;
import be.superjoran.mint.domain.Person;
import be.superjoran.mint.domain.searchresults.BankAccountCandidate;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Service;

import java.util.List;

import static java.util.Objects.requireNonNull;

@Service
public class BankAccountServiceImpl extends DomainObjectCrudServiceSupport<BankAccount> implements BankAccountService {
    private final BankAccountDao bankAccountDao;

    public BankAccountServiceImpl(BankAccountDao bankAccountDao) {
        this.bankAccountDao = bankAccountDao;
    }

    @Override
    protected CrudRepository<BankAccount, String> getDao() {
        return this.bankAccountDao;
    }

    @Override
    public List<BankAccount> findAllByOwner(Person administrator) {
        return this.bankAccountDao.findAllByOwner(administrator);
    }

    @Override
    public BankAccount createOrUpdate(BankAccountCandidate bankAccountCandidate) {
        return this.save(new BankAccount(bankAccountCandidate.getOwner(), requireNonNull(bankAccountCandidate.getBank()), requireNonNull(bankAccountCandidate.getNumber()), bankAccountCandidate.getName()));
    }
}

File: src/main/java/be/superjoran/mint/services/BankAccountServiceImpl.kt
package be.superjoran.mint.services

import be.superjoran.common.model.DomainObjectCrudServiceSupport
import be.superjoran.mint.dao.BankAccountDao
import be.superjoran.mint.domain.Bank
import be.superjoran.mint.domain.BankAccount
import be.superjoran.mint.domain.Person
import be.superjoran.mint.domain.searchresults.BankAccountCandidate
import org.springframework.stereotype.Service
import java.util.Objects.requireNonNull
import javax.validation.Validator

@Service
class BankAccountServiceImpl(
        override val dao: BankAccountDao,
        override val validator: Validator
) : DomainObjectCrudServiceSupport<BankAccount>(), BankAccountService {

    override fun findAllByOwner(administrator: Person): List<BankAccount> {
        return this.dao.findAllByOwner(administrator)
    }

    override fun create(bankAccountCandidate: BankAccountCandidate): BankAccount {
        return this.save(BankAccount(bankAccountCandidate.owner, requireNonNull<Bank>(bankAccountCandidate.bank), requireNonNull<String>(bankAccountCandidate.number), bankAccountCandidate.name))
    }
}
